# CAR-SALES dataset

Dữ liệu `car-sales.csv`: bộ dữ liệu chứa thông tin các ôtô cũ ở một cửa hàng bán ô tô cũ. 

Bạn hãy thực hiện các yêu cầu sau:


In [1]:
import pandas as pd
import numpy as np

### 1. Đọc dữ liệu và cho biết dữ liệu có bao nhiêu dòng, bao nhiêu cột

Đáp án: (10 dòng, 5 cột)

In [2]:
cars = pd.read_csv('data/car-sales.csv')
cars

Manufacturer Colour  Odometer  Doors    Price
0       Toyota  White    150043      4   4000.0
1        Honda    Red     87899      4   5000.0
2       Toyota   Blue     32549      3   7000.0
3          BMW  Black     11179      5  22000.0
4       Nissan  White    213095      4   3500.0
5       Toyota  Green     99213      4   4500.0
6        Honda   Blue     45698      4   7500.0
7        Honda   Blue     54738      4   7000.0
8       Toyota  White     60000      4   6250.0
9       Nissan  White     31600      4   9700.0

### 2. Cho biết thuộc tính nào là thuộc tính phân loại, thuộc tính nào là thuộc tính định lượng

In [3]:
dinhLuong = []
dinhTinh = []
for column in cars.columns:
        if (cars[column].dtype == 'object'):
                dinhTinh.append(column)
        else:
                dinhLuong.append(column)
print(f'Thuộc tính định tính là: ', dinhTinh)
print(f'Thuộc tính định lượng là: ', dinhLuong)

Thuộc tính định tính là:  ['Manufacturer', 'Colour']
Thuộc tính định lượng là:  ['Odometer', 'Doors', 'Price']


### 3. Trong dữ liệu có thông tin của bao nhiêu xe?

Đáp án: 10 xe

In [6]:
cars['Manufacturer'].count()

10

### 4. Đếm số xe theo theo hãng

Đáp án: Toyota: 10 xe, Honda: 3 xe, Nissan: 2 xe, BMW: 1 xe

In [7]:
cars['Manufacturer'].value_counts()

Toyota    4
Honda     3
Nissan    2
BMW       1
Name: Manufacturer, dtype: int64

### 5. Đếm số xe theo màu


In [8]:
cars['Colour'].value_counts()

White    4
Blue     3
Red      1
Black    1
Green    1
Name: Colour, dtype: int64

### 6. Tổng giá trị các xe trong cửa hàng

Đáp án: 76450 USD


In [11]:
cars['Price'].sum()

76450.0

### 7. Top 3 xe có giá cao nhất


In [12]:
cars.sort_values(by='Price', ascending=False).head(3)

Manufacturer Colour  Odometer  Doors    Price
3          BMW  Black     11179      5  22000.0
9       Nissan  White     31600      4   9700.0
6        Honda   Blue     45698      4   7500.0

### 8. Top 3 xe đã chạy nhiều cây số (km)nhất (Odometer)

Đáp án:
- Xe Nissan, White, 213095 km
- Xe Toyota, White, 150043 km
- Xe Toyota, Green, 99213 km

In [13]:
cars.sort_values(by='Odometer', ascending=False).head(3)

Manufacturer Colour  Odometer  Doors   Price
4       Nissan  White    213095      4  3500.0
0       Toyota  White    150043      4  4000.0
5       Toyota  Green     99213      4  4500.0

### 9. Khảo sát xe của hãng Honda

- Tính tỷ lệ xe Honda trong dữ liệu của cửa hàng (0.3)
- Cho biết giá trung bình của xe hãng Honda của cửa hàng (6500)
- Cho biết các xe của hãng Honda trong cửa hàng có những màu sắc nào (Red, Blue)

In [19]:
print(cars['Manufacturer'][cars.Manufacturer == 'Honda'].count()/cars['Manufacturer'].count()) 
print(cars['Price'][cars.Manufacturer == 'Honda'].mean())
print(cars['Colour'][cars.Manufacturer == 'Honda'].value_counts())

0.3
6500.0
Blue    2
Red     1
Name: Colour, dtype: int64


# 10. Tính toán các giá trị sau của  thuộc tính giá ('Price')

- Giá trung bình các xe (7645)
- Giá trị trung vị các xe (6625)
- Miền giá trị của xe (18500)
- Miền phân vị của xe (2750)
- Điểm phân vị 0.4 và 0.6 của giá ([5750, 7000])

In [27]:
print(cars['Price'].mean())
print(cars['Price'].median())
print(cars['Price'].max() - cars['Price'].min())
print(cars['Price'].quantile(0.75)-cars['Price'].quantile(0.25))
print(cars['Price'].quantile([0.4, 0.6]))

7645.0
6625.0
18500.0
2750.0
0.4    5750.0
0.6    7000.0
Name: Price, dtype: float64


---

# TIPS dataset

Dữ liệu chứa trong file `tips.csv` chứa thông tin tiền tip của khách hàng tại một nhà hàng. Dữ liệu có các thông tin sau:

- `total_bill`: tổng hoá đơn (USD)
- `tip`: tiền tip/ tiền boa (USD)
- `sex`: giới tính người thanh toán hóa đơn
- `smoker`: có hút thuốc hay không
- `day`: ngày trong tuần
- `time`: thời gian dùng bữa(buổi sáng, buổi trưa, buổi tối)
- `size`: số lượng người tham gia

Bạn hãy thực hiện thống kê mô tả để tìm hiểu về hành vi tip của khách hàng tại nhà hàng

1. Đọc dữ liệu


In [28]:
tips = pd.read_csv('data/tips.csv')
tips

Unnamed: 0  total_bill   tip     sex smoker   day    time  size
0             0       16.99  1.01  Female     No   Sun  Dinner     2
1             1       10.34  1.66    Male     No   Sun  Dinner     3
2             2       21.01  3.50    Male     No   Sun  Dinner     3
3             3       23.68  3.31    Male     No   Sun  Dinner     2
4             4       24.59  3.61  Female     No   Sun  Dinner     4
..          ...         ...   ...     ...    ...   ...     ...   ...
239         239       29.03  5.92    Male     No   Sat  Dinner     3
240         240       27.18  2.00  Female    Yes   Sat  Dinner     2
241         241       22.67  2.00    Male    Yes   Sat  Dinner     2
242         242       17.82  1.75    Male     No   Sat  Dinner     2
243         243       18.78  3.00  Female     No  Thur  Dinner     2

[244 rows x 8 columns]

2. Xoá cột `Unnamed: 0`


In [31]:
tips = tips.drop(columns='Unnamed: 0', )
tips

total_bill   tip     sex smoker   day    time  size
0         16.99  1.01  Female     No   Sun  Dinner     2
1         10.34  1.66    Male     No   Sun  Dinner     3
2         21.01  3.50    Male     No   Sun  Dinner     3
3         23.68  3.31    Male     No   Sun  Dinner     2
4         24.59  3.61  Female     No   Sun  Dinner     4
..          ...   ...     ...    ...   ...     ...   ...
239       29.03  5.92    Male     No   Sat  Dinner     3
240       27.18  2.00  Female    Yes   Sat  Dinner     2
241       22.67  2.00    Male    Yes   Sat  Dinner     2
242       17.82  1.75    Male     No   Sat  Dinner     2
243       18.78  3.00  Female     No  Thur  Dinner     2

[244 rows x 7 columns]

3. Cho biết dữ liệu có bao nhiêu dòng, bao nhiêu thuộc tính.


In [34]:
print(tips.shape[0])
print(tips.shape[1])


244
7


4. Trong các thuộc tính, thuộc tính nào là thuộc tính phân loại, thuộc tính nào là thuộc tính định lượng


In [40]:
dinhTinh = []
dinhLuong = []
for i in tips.columns:
        if (tips[i].dtype == 'object'):
                dinhTinh.append(i)
        else:
                dinhLuong.append(i)
print(dinhTinh)
print(dinhLuong)

['sex', 'smoker', 'day', 'time']
['total_bill', 'tip', 'size']


5. Có bao nhiêu nam, bao nhiêu nữ. Tỷ lệ bao nhiêu?


In [46]:
soLuongNam = tips['sex'][tips.sex == 'Male'].value_counts()
print(soLuongNam)
soLuongNu = tips['sex'][tips.sex == 'Female'].value_counts()
print(soLuongNu)
print(f'Tỷ lệ nam là: ', soLuongNam/tips['sex'].count())
print(f'Tỷ lệ nữ là: ', soLuongNu/tips['sex'].count())

Male    157
Name: sex, dtype: int64
Female    87
Name: sex, dtype: int64
Tỷ lệ nam là:  Male    0.643443
Name: sex, dtype: float64
Tỷ lệ nữ là:  Female    0.356557
Name: sex, dtype: float64


6. Có bao nhiêu người hút thuốc. Tỷ lệ bao nhiêu?


In [47]:
soNguoiHutThuoc = tips['smoker'][tips.smoker == 'Yes'].value_counts()
print(soNguoiHutThuoc)
print(f'Tỷ lệ: ', soNguoiHutThuoc/tips['smoker'].count())

Yes    93
Name: smoker, dtype: int64
Tỷ lệ:  Yes    0.381148
Name: smoker, dtype: float64


7. Ngày nào trong tuần khách hàng hay đến dùng bữa nhất. Tỷ lệ bao nhiêu?


In [54]:
dayOfWeek = tips['day'].value_counts().head(1)
print(dayOfWeek)
print(f'Tỷ lệ là: ', dayOfWeek/tips['day'].count())

Sat    87
Name: day, dtype: int64
Tỷ lệ là:  Sat    0.356557
Name: day, dtype: float64


8. Nhà hàng phục vụ bữa ăn nào nhiều nhất (Breakfast, Lunch, Dinner)?


In [56]:
buaAnNhieuNhat = tips['time'].value_counts().head(1)
print(buaAnNhieuNhat)

Dinner    176
Name: time, dtype: int64


9. Nhóm khách hàng dùng bữa tại nhà hàng nhiều nhất là bao nhiêu người?


In [59]:
nhomNhieuNhat = tips['size'].value_counts().head(1)
print(nhomNhieuNhat)

2    156
Name: size, dtype: int64


10. Trung bình trong một bữa ăn nhà hàng phục vụ mấy khách?

In [60]:
trungBinh = tips.groupby('time')['size'].mean()
print(trungBinh)

time
Dinner    2.630682
Lunch     2.411765
Name: size, dtype: float64


---